subprocess_utils.hpp

```diff
@@ -0,0 +1,219 @@
+#pragma once
+
+#include <string>
+#include <iostream>
+
+// unix process stuff
+#include <signal.h>
+#include <unistd.h>
+#include <sys/wait.h>
+#include <sys/prctl.h>
+#include <cstring>
+
+
+namespace subprocess {
+    
+    /**
+     * A TwoWayPipe that allows reading and writing between two processes
+     * must call initialize before being passed between processes or used
+     * */
+class TwoWayPipe {
+	private:
+		//[0] is the output end of each pipe and [1] is the input end of each pipe
```

Comment:
careful about consistent use of indentation - this uses both tabs and spaces, and isn't matched

Reply:
Yeah I worked so hard on the first commit I totally forgot after that

Reply:
i'm currently writing up some travis/tests so you should be able to rely on travis to do things hopefully soon, and develop solely in windows (i'll need your help for the windows build stuff maybe though)